### DISCUSSION TITLE
Why is your README.md file and official website not friendly at all, and you don’t know how to use it at all

### DISCUSSION THREAD

**Discussion by poohlaha** [hyperium/hyper#3244] [General]
I used hyper version 1.0 for the first time. I checked the official website and the README.md document, but I didn’t know how to use it. Then I checked the examples. As a result, the compiler reported a lot of errors, and I couldn’t start. Please complete the README.md and official documents. Add technical documentation manual, thank you~

**Comment by poohlaha** [comment]
i have given up

**Comment by fergus-hou** [comment]
Maybe you should give the specific problem you have encountered, otherwise it will be difficult for others to help you solve the problem.
Also, you can consider using a web framework already based on hyper 1.0, eg: [salvo](https://github.com/salvo-rs/salvo).

**Comment by seanmonstar** [comment]
I'm sorry you had a difficult time. We always want to improve the documentation and onboarding. Would you mind sharing what you tried to do, and what part was difficult?

**Comment by guradia** [comment]
There is some disconnect between the examples in the git folder and the website's guides.

When following the guide at https://hyper.rs/guides/1/server/hello-world/, you will get a working example.
But since you have to piece it together, you may be tempted to follow the link at the bottom to the "full example" (https://github.com/hyperium/hyper/blob/master/examples/hello.rs)

Which is not the same example. It has at least 2 problems:

1. You have to locate the right dependencies (which differ from the guide and you will probably land on the Cargo.toml of the library)
2. There is this whole `benches/` directory which makes this example something very different than the guide
